### DISCUSSION TITLE
[Question][Hw0-2] Sigma function of calculator.ml

### DISCUSSION THREAD

**Discussion by adzaky15** [prosyslab-classroom/cs348-information-security#649] [Q&A]
Name: Akhdan Dzaky Maulana

Hello!
I'm doing calculator.ml, and I got stuck on the Sigma(e1, e2, e3) -> part.

Should I add "env" map to the calculator? Or should I find a way to convert e3 into a function?

**Comment by pingpingy1** [comment]
I think you can write a helper function that takes the environment into account, and then use it inside the calculator function.

**Comment by ICubE01** [comment]
There would be two solutions:

1. Make a helper function that has an environment argument `calc_aux (expr) (env)` and save the value of `X` in the environment.
2. Since only one variable named `X` exists, substitute all `X` in e3, except for internal sigma.

First one is more general solution, but second one is easier to implement.

**Comment by bongjunj** [comment]
I did it iterating all possible values of X when evaluating `Sigma` expression. :)

**Comment by KihongHeo** [comment]
@adzaky15 does the above comments solve your problem? Please close this issue if so.
